policy-design-lab/pdl-frontend
# #268: The EQIP IRA page

This PR is for #268 and includes content for the IRA page, with the EQIP section as the first step. You can review it on the dev site: https://policydesignlab-dev.ncsa.illinois.edu/ira. I will mark this release as _1.0_ as we discussed before, but I am open to your thoughts.

Please note that in this PR, I commented out some code while working on the IRA prototype, such as the slider and some min/max prediction calculations. _I want to keep them until the entire IRA page is finished_ so that if collaborators request their re-inclusion, I can easily uncomment these codes instead of recreating the component.

### How to Test

1. Navigate to https://policydesignlab-dev.ncsa.illinois.edu/ira and confirm the model is populated correctly. The model shape could be different depending on your screen shape.

2. Check the map to ensure the data is correctly displayed.

3. Scroll to the bottom of the page to check that the table is correctly displayed, including both the dollar table and the corresponding nationwide percentage table.

4. When you click the "Export this Table to CSV" button, you should be able to download a CSV file named 'pdl-data' for this table:
<img width="2378" alt="Screenshot 2024-06-28 at 9 33 27 AM" src="https://github.com/policy-design-lab/pdl-frontend/assets/92752107/3388db86-bdbb-4aba-9e5f-4f88c3e9ce2c">

5. Now navigate to the right side of the page and select random practices instead of Total. You should see the map and tables reflect the updated information. Repeat steps 2 to 4:
<img width="560" alt="Screenshot 2024-06-28 at 9 33 41 AM" src="https://github.com/policy-design-lab/pdl-frontend/assets/92752107/e6898aa6-9605-4433-9480-59ac0d9bcc3c">

6. Select 'Total' from the "Select Practice" list. You should see the total data returned, and all practices cleared out:
<img width="663" alt="Screenshot 2024-06-28 at 9 33 51 AM" src="https://github.com/policy-design-lab/pdl-frontend/assets/92752107/0148723b-b9a6-4eed-a8e9-3d48154b38b6">

7. Now turn the "2024-2031 Prediction" switch on. You should see all data become the predicted data. The map and tables should reflect the changes:
<img width="432" alt="Screenshot 2024-06-28 at 9 34 06 AM" src="https://github.com/policy-design-lab/pdl-frontend/assets/92752107/8b5c4023-cb86-4094-980d-184e1e266430">

8. Repeat steps 2 to 6 for the predicted data.
